### DISCUSSION TITLE
Setting up on Windows

### DISCUSSION THREAD

**Discussion by bug-author** [AntonOsika/gpt-engineer#187] [Q&A] [answered]
Hi. I believe the setup for Windows (without WSL) might be a little bit different from the one given in README.md. If anyone has been able to do so then can they please give some instructions how they did it?

**Comment by mitchd319** [comment, marked as answer]
For a Windows user, you just need to perform the setup that would otherwise be performed by the Makefile.

1. Create a virtual environment named "venv". See https://www.youtube.com/watch?v=APOPm01BVrk for instructions
2. Run: pip install --upgrade pip
3. Run: pip install -r requirements.txt
4. If you plan on contributing, run: pre-commit install

**Reply by bug-author** [reply]
Thanks! When I posted this question I wasn't willing to go through WSL installation haha. But then last last week I had to do it at work so I just went with setting up GPT-Engineer on WSL. I'll mark this as an answer (I haven't tested it) :)

**Reply by famousgit** [reply]
there is no requirements.txt in the project

**Reply by famousgit** [reply]
the makefile is not working under windows and the requirements.txt is removed. Can somebody give some help ?

**Reply by famousgit** [reply]
Nevermind. i have run it succesfully under windows without requirements.txt

**Reply by ataxy** [reply]
@famousgit it would be coolto know how?

**Comment by CharlieAlphaFox** [comment]
Im getting this error on Win when specifying the open ai key:

'export' is not recognized as an internal or external command,
operable program or batch file.

**Reply by CharlieAlphaFox** [reply]
On win10 you need 'set' as 'export' is for Linux

**Reply by danielenriquez59** [reply]
you can also manually set your environment variable for OPENAI_API_KEY

**Reply by geewhizbang** [reply]
Can we rewrite the instructions to actually say this for windows, huh? I had to throw out a key because this wasn't clear. I eventually guessed this, but geesh. I did put the new key into a private place so I won't lose this one.

It would also be helpful to break out the steps so they aren't ganged together. I missed the bit about creating the "prompt" file, and it would be helpful if it actually suggested what you could put into it to make it work. For newbies, you have to assume that we really don't know what we are doing yet.
